setup.py

```diff
@@ -23,7 +24,12 @@
     "numpy": "1.18.0",
     "pandas": "1.0.0",
     "scikit-learn": "0.23.0",
-    "statsmodels": "0.12.0"
+    "statsmodels": "0.12.0",
+}
+EXTRAS_REQUIRE = {
```

Comment:
What's the benefit of adding them here as EXTRAS_REQUIRE? I know that you could now run `pip install sktime[series_as_features]`. 

I'm a little wary of specifying the extra requirements in another place, we already need to keep .binder/requirements.txt and build_tools/requirements.txt in sync.

Reply:
Initially I was going to do `pip install sktime[all_extras]` instead of specifying each individual soft dependency before I realised there were so many. I know it's another place to keep track and sync dependencies, but personally, I think it would make it easier for new users to `sktime`. Rather than having to hunt through the `sktime` module itself to find additional requirements.txt, or run various modules and consistently get `ModuleNotFoundError`s, they can simply run a standard `pip install sktime[all_extras]` command. We could even add `pip install sktime[all_extras]` to the `ModuleNotFoundError` when soft dependencies aren't installed

Reply:
Okay, let's do that! We could go through the CI scripts to use `pip install -e .[all_extras]` there too, I'm happy to look into this if you don't have the time, but would wait for you to finish your work.

Reply:
We need to add: numba (sorry, forgot about it, see build_tools/requirements.txt)